nerdlandbot/commands/space_launches.py

```diff
@@ -0,0 +1,117 @@
+            async with aiohttp.ClientSession() as session:
+                async with session.get(full_url, headers = {"accept":"application/json"}) as resp: 
+                    if resp.status == 200:
+                        msg = await resp.text()
+                        with open(os.path.join(THE_SPACE_DEVS_LOCAL_CACHE_FOLDER, THE_SPACE_DEVS_LOCAL_CACHE_SPACE_LAUNCHES_FILE + '.json'),"w") as file:
+                            file.write(msg)
+                        with open(os.path.join(THE_SPACE_DEVS_LOCAL_CACHE_FOLDER, THE_SPACE_DEVS_LOCAL_CACHE_SPACE_LAUNCHES_FILE + '.time'),"w") as file:
+                            timestamp = datetime.now().strftime(THE_SPACE_DEVS_TIMESTAMP_FORMAT)
+                            file.write(timestamp)
+                        await self.parse_and_send_results(msg)
+                    elif resp.status == 429:
+                        await ctx.send('Too many requests. Wait a couple of minutes and try again.')
+                    else:
+                        await ctx.send('Call to the space devs failed. Response status: '+ str(resp.status))
+                    return
+        else:
+            with open(os.path.join(THE_SPACE_DEVS_LOCAL_CACHE_FOLDER, THE_SPACE_DEVS_LOCAL_CACHE_SPACE_LAUNCHES_FILE + '.json'),"r") as file:
+                msg = file.read()
+            await self.parse_and_send_results(msg)
+
+    def should_call_the_api (self):
+        if os.path.isfile(os.path.join(THE_SPACE_DEVS_LOCAL_CACHE_FOLDER, THE_SPACE_DEVS_LOCAL_CACHE_SPACE_LAUNCHES_FILE + '.time')):
+            with open(os.path.join(THE_SPACE_DEVS_LOCAL_CACHE_FOLDER, THE_SPACE_DEVS_LOCAL_CACHE_SPACE_LAUNCHES_FILE + '.time'),"r") as file:
+                call_timestamp = datetime.strptime(file.readline(),THE_SPACE_DEVS_TIMESTAMP_FORMAT)
+            now_timestamp = datetime.now()
+            return (now_timestamp - call_timestamp).total_seconds() > 3600
+        else:
+            return True
+        
+    async def parse_and_send_results(self, json_string):    # extracted for testing purposes
+        try:
+            dom = json.loads(json_string)
+        except json.JSONDecodeError:
+            await self.ctx.send('Could not parse the response from space devs.')
+            return
+        embed = self.main_info_embed()
+        for index, result in enumerate (dom['results']):
+            await self.get_embed_field_for_upcominglaunch(index, result, embed)
+        await self.ctx.send(embed = embed)
+        return
+
+    def main_info_embed (self):
+        result = discord.Embed(
+                                title="Upcoming launches", 
+                                url=THE_SPACE_DEVS_HOME_URL, 
+                                description="Provided by the space devs api. Timestamps = UTC",
+                                color=discord.Color.blue()
+                            )
+        return result
+
+    async def get_embed_field_for_upcominglaunch(self, index, result_json, embed):
```

Comment:
Is it necessary that this is an async function or can this be a regular def?

Reply:
Removed all async shizzle from functions that create the embedding stuff